node/heartbeat/monitor.go

```diff
@@ -252,6 +252,8 @@ func (m *Monitor) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedPe
 		return err
 	}
 
+	//TODO check if hardfork trigger can be set otherwise (not requiring a pk that will actually send the trigger
```

Comment:
it is not whenever it wants - it is the PK managed by an off-chain governance, people can opt out of this. They can set their own PK to activate this trigger. Later on this trigger, can be done in on-chain governance.

Reply:
Agree, TODO will remain as to remind us to re-think of a better solution (if we found one)